otava/change_point_divisive/base.py

```diff
@@ -0,0 +1,109 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from dataclasses import dataclass, fields
+from typing import Generic, List, Optional, TypeVar
+
+from numpy.typing import NDArray
+
+
+@dataclass
+class CandidateChangePoint:
+    '''Candidate for a change point. The point that maximizes Q-hat function on [start:end+1] slice'''
+    index: int
```

Comment:
Could add a mention that indexes are always the original index from the full series [0,T] and not zero-based for each interval. (In math terms, `start < index <= end`,  never `0 < index <= end-start`

Btw, what is the +1 from? Perhaps you meant to say as in the python `range(start, stop) = range(start, end+1)` where the second parameter is not included in the output.

If this was the reason, I would rather say the interval is between start and end and the change point is in `]start, end]` 

A change point at index 0 is impossible, because a single point just cannot change But this IMO follows logically, it is not a pre-condition. But could be added as an invariant perhaps?